src/main/java/jcifs/smb1/smb1/SmbSession.java

```diff
@@ -0,0 +1,476 @@
+        SmbException ex = null;
+        SmbComSessionSetupAndX request;
+        SmbComSessionSetupAndXResponse response;
+        byte[] token = new byte[0];
+        int state = 10;
+
+        while (connectionState != 0) {
+            if (connectionState == 2 || connectionState == 3) // connected or disconnecting
+                return;
+            try {
+                transport.wait();
+            } catch (InterruptedException ie) {
+                throw new SmbException(ie.getMessage(), ie);
+            }
+        }
+        connectionState = 1; // trying ...
+
+        try {
+            transport.connect();
+
+            /*
+             * Session Setup And X Request / Response
+             */
+    
+            if( transport.log.level >= 4 )
+                transport.log.println( "sessionSetup: accountName=" + auth.username + ",primaryDomain=" + auth.domain );
+    
+            /* We explicitly set uid to 0 here to prevent a new
+             * SMB_COM_SESSION_SETUP_ANDX from having it's uid set to an
+             * old value when the session is re-established. Otherwise a
+             * "The parameter is incorrect" error can occur.
+             */
+            uid = 0;
+    
+            do {
+                switch (state) {
+                    case 10: /* NTLM */
+                        if (auth != NtlmPasswordAuthentication.ANONYMOUS &&
+                                transport.hasCapability(SmbConstants.CAP_EXTENDED_SECURITY)) {
+                            state = 20; /* NTLMSSP */
+                            break;
+                        }
+    
+                        request = new SmbComSessionSetupAndX( this, andx, auth );
+                        response = new SmbComSessionSetupAndXResponse( andxResponse );
+    
+                        /* Create SMB signature digest if necessary
+                         * Only the first SMB_COM_SESSION_SETUP_ANX with non-null or
+                         * blank password initializes signing.
+                         */
+                        if (transport.isSignatureSetupRequired( auth )) {
+                            if( auth.hashesExternal && NtlmPasswordAuthentication.DEFAULT_PASSWORD != NtlmPasswordAuthentication.BLANK ) {
+                                /* preauthentication
+                                 */
+                                transport.getSmbSession( NtlmPasswordAuthentication.DEFAULT ).getSmbTree( LOGON_SHARE, null ).treeConnect( null, null );
+                            } else {
+                                byte[] signingKey = auth.getSigningKey(transport.server.encryptionKey);
+                                request.digest = new SigningDigest(signingKey, false);
+                            }
+                        }
+    
+                        request.auth = auth;
+    
+                        try {
+                            transport.send( request, response );
+                        } catch (SmbAuthException sae) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 393.